internal/pkg/manifest/validate.go

```diff
@@ -0,0 +1,410 @@
+	return b.BuildArgs.Validate()
+}
+
+// Validate returns if DockerBuildArgs is configured correctly.
+func (*DockerBuildArgs) Validate() error {
+	return nil
+}
+
+// Validate returns if ContainerHealthCheck is configured correctly.
+func (*ContainerHealthCheck) Validate() error {
+	return nil
+}
+
+// Validate returns if ImageOverride is configured correctly.
+func (i *ImageOverride) Validate() error {
+	var err error
+	if err = i.EntryPoint.Validate(); err != nil {
+		return fmt.Errorf(`validate "entrypoint": %w`, err)
+	}
+	if err = i.Command.Validate(); err != nil {
+		return fmt.Errorf(`validate "command": %w`, err)
+	}
+	return nil
+}
+
+// Validate returns if EntryPointOverride is configured correctly.
+func (*EntryPointOverride) Validate() error {
+	return nil
+}
+
+// Validate returns if CommandOverride is configured correctly.
+func (*CommandOverride) Validate() error {
+	return nil
+}
+
+// Validate returns if RoutingRule is configured correctly.
+func (r *RoutingRule) Validate() error {
+	var err error
+	if !r.HealthCheck.IsEmpty() {
+		if err = r.HealthCheck.Validate(); err != nil {
+			return fmt.Errorf(`validate "healthcheck": %w`, err)
+		}
+	}
+	if !r.Alias.IsEmpty() {
+		if err = r.Alias.Validate(); err != nil {
+			return fmt.Errorf(`validate "alias": %w`, err)
+		}
+	}
+	if r.TargetContainer != nil && r.TargetContainerCamelCase != nil {
+		return &errFieldMutualExclusive{
+			firstField:  "target_container",
+			secondField: "targetContainer",
+		}
+	}
+	return nil
+}
+
+// Validate returns if HealthCheckArgsOrString is configured correctly.
+func (h *HealthCheckArgsOrString) Validate() error {
+	return h.HealthCheckArgs.Validate()
+}
+
+// Validate returns if HTTPHealthCheckArgs is configured correctly.
+func (*HTTPHealthCheckArgs) Validate() error {
+	return nil
+}
+
+// Validate returns if Alias is configured correctly.
+func (*Alias) Validate() error {
+	return nil
+}
+
+// Validate returns if TaskConfig is configured correctly.
+func (t *TaskConfig) Validate() error {
+	var err error
+	if err = t.Platform.Validate(); err != nil {
+		return fmt.Errorf(`validate "platform": %w`, err)
+	}
+	if err = t.Count.Validate(); err != nil {
+		return fmt.Errorf(`validate "count": %w`, err)
+	}
+	if err = t.ExecuteCommand.Validate(); err != nil {
+		return fmt.Errorf(`validate "exec": %w`, err)
+	}
+	if err = t.Storage.Validate(); err != nil {
+		return fmt.Errorf(`validate "storage": %w`, err)
+	}
+	return nil
+}
+
+// Validate returns if PlatformArgsOrString is configured correctly.
+func (p *PlatformArgsOrString) Validate() error {
+	if !p.PlatformArgs.isEmpty() {
+		return p.PlatformArgs.Validate()
+	}
+	return nil
+}
+
+// Validate returns if PlatformArgsOrString is configured correctly.
+// TODO: add validation once "feat/pencere" is merged.
+func (*PlatformArgs) Validate() error {
+	return nil
+}
+
+// Validate returns if Count is configured correctly.
+func (c *Count) Validate() error {
+	if !c.AdvancedCount.IsEmpty() {
```

Comment:
Same here. Since `count` is composite, can this be 
```
return c.AdvancedCount.Validate()
```
just like `BuildArgsOrString`?

Reply:
Discussed offline